### DISCUSSION TITLE
Refactor `withdrawMultiple` to in include `block.timestamp`

### DISCUSSION THREAD

**Discussion by smol-ninja** [sablier-labs/flow#53] [General]
### Motivation

- An array of `time` cannot trigger a full withdrawal due to unpredictability of `block.timestamp`
- That means, it wouldn't work if the intent is to withdraw the "full" amount from multiple streams which imo is going to be more common.

That is why I suggest to remove `uint40[] calldata times` from it and use `block.timestamp` instead.

```solidity
function withdrawMultiple(uint256[] calldata streamIds) external;
```

instead of 

```solidity
function withdrawMultiple(uint256[] calldata streamIds, uint40[] calldata times) external;
```

**Side note**: If a user intents to include `time` in `withdrawMultiple`, they can do it programatically using `withdraw` function.

**Comment by andreivladbrg** [comment]
> An array of time cannot trigger a full withdrawal due to unpredictability of block.timestamp

i don't think i understand this point

---

why don't we add `withdrawMaxMultiple` instead and keep:

```solidity
function withdrawMultiple(uint256[] calldata streamIds, uint40[] calldata times) external;
```

**Reply by smol-ninja** [reply]
> i don't think i understand this point

By passing `time` value as the current time, there will be a residual amount left in the stream proportional to `block.timestamp - inputTime`.

> why don't we add withdrawMaxMultiple instead and keep

The most useful functions will be called `withdrawMax` and `withdrawMaxMultiple`. I was suggesting to call the most useful functions as `withdraw` and `withdrawMultiple`.

**Reply by smol-ninja** [reply]
We agree to define a new function called `withdrawMaxMultiple`.

**Comment by smol-ninja** [comment]
Issue created:

- https://github.com/sablier-labs/v2-open-ended/issues/73
